server/src/main/java/org/opensearch/action/admin/indices/cache/clear/TransportClearIndicesCacheAction.java

```diff
@@ -129,11 +146,25 @@ protected ShardsIterator shards(ClusterState clusterState, ClearIndicesCacheRequ
 
     @Override
     protected ClusterBlockException checkGlobalBlock(ClusterState state, ClearIndicesCacheRequest request) {
+        ensureExclusiveClearFileCacheRequest(request);
```

Comment:
Question here please: does metadata block prevents file cache from cleaning? I don't see any obstacles there (file cache pruning does not refer to metadata), apologies if I am missing something obvious here.

Reply:
The block would not affect the filecache from being cleaned, but we do have write blocks on searchable snapshot indices. This causes this exception to be thrown: https://github.com/opensearch-project/OpenSearch/blob/ab0f2620f7914494cc902a3e3cbe23b68ab2c070/server/src/main/java/org/opensearch/action/admin/indices/cache/clear/TransportClearIndicesCacheAction.java#L162
from here - https://github.com/opensearch-project/OpenSearch/blob/9d6255c61748ffdf0afde11a3724d00d4b1f1d05/server/src/main/java/org/opensearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java#L305-L309C3

Another alternative is to modify `checkRequestBlock` and filter out searchable snapshot indices from the list of concrete indices as follows - 

```
@Override
    protected ClusterBlockException checkRequestBlock(ClusterState state, ClearIndicesCacheRequest request, String[] concreteIndices) {
        String[] filteredIndices = Arrays.stream(concreteIndices)
            .filter(
                index -> !indicesService.indexService(state.routingTable().index(index).getIndex()).getIndexSettings().isRemoteSnapshot()
            )
            .toArray(String[]::new);
        return state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA_WRITE, filteredIndices);
    }
```

Reply:
Thanks a lot @kotwanikunal , I believe going this extra mile would definitely make API more coherent. The one particular case is clear all caches (no request parameters): we won't be able to do that (since this is == combination of file cache with all other caches). @andrross do you have an opinion on that?

Reply:
This API doesn't make changes to metadata for any of the caches. I am wondering why should Metadata_Write block on the index should restrict any of the cache clean up. @kotwanikunal . Should this have been relaxed for all caches and indices?

Reply:
I can walk down the call path to verify if it does not affect the metadata in cases other than the filecache. In that scenario, we surely can get rid of the metadata write block check.

Reply:
I don't know why the metadata write block prevents clearing caches. My opinion is that clearing caches should be allowed for any index that allows reading (including read only).

Reply:
I walked through the cache clearing paths and it does not look like it affects the metadata in any form. I'll move ahead with changing it down to a metadata read block instead. 

Code walkthrough dump: 

Query Cache: 
- https://github.com/opensearch-project/OpenSearch/blob/56a2beda80344ce84f2f128476f1d6c0df65f799/server/src/main/java/org/opensearch/index/IndexService.java#L1246-L1259
- https://github.com/opensearch-project/OpenSearch/blob/7b75fb425f9df2b96049dac8c7dc13273f5026fd/server/src/main/java/org/opensearch/index/cache/query/IndexQueryCache.java#L63-L66
- https://github.com/opensearch-project/OpenSearch/blob/c0a8cf8823341cf92b932fe3d12909d427090a2f/server/src/main/java/org/opensearch/indices/IndicesQueryCache.java#L196-L209

FieldsData: 
- https://github.com/opensearch-project/OpenSearch/blob/56a2beda80344ce84f2f128476f1d6c0df65f799/server/src/main/java/org/opensearch/index/IndexService.java#L1246-L1259
- https://github.com/opensearch-project/OpenSearch/blob/7b75fb425f9df2b96049dac8c7dc13273f5026fd/server/src/main/java/org/opensearch/index/fielddata/IndexFieldDataService.java#L108-L132
- https://github.com/opensearch-project/OpenSearch/blob/0ca51a774211184835c4825dfeff38b23198352e/server/src/main/java/org/opensearch/indices/fielddata/cache/IndicesFieldDataCache.java#L227-L252
- Listener: https://github.com/opensearch-project/OpenSearch/blob/56a2beda80344ce84f2f128476f1d6c0df65f799/server/src/main/java/org/opensearch/index/IndexService.java#L835-L843
- https://github.com/opensearch-project/OpenSearch/blob/7b75fb425f9df2b96049dac8c7dc13273f5026fd/server/src/main/java/org/opensearch/index/fielddata/ShardFieldData.java#L92C1-L104

Request Cache:
- https://github.com/opensearch-project/OpenSearch/blob/86f8dd65a45810ae671ee1f607a2d5287649d66b/server/src/main/java/org/opensearch/indices/IndicesRequestCache.java#L132C1-L135
- https://github.com/opensearch-project/OpenSearch/blob/86f8dd65a45810ae671ee1f607a2d5287649d66b/server/src/main/java/org/opensearch/indices/IndicesRequestCache.java#L333-L361
- Listener: https://github.com/opensearch-project/OpenSearch/blob/0ca51a774211184835c4825dfeff38b23198352e/server/src/main/java/org/opensearch/indices/AbstractIndexShardCacheEntity.java#L69-L71

Reply:
Raised a PR here: https://github.com/opensearch-project/OpenSearch/pull/7538